iwyu_output.h

```diff
@@ -216,6 +216,13 @@ class OneIncludeOrForwardDeclareLine {
   const clang::NamedDecl* fwd_decl_;
 };
 
+struct NamespaceDeclUse {
+  clang::SourceLocation used_loc;
+  const clang::NamespaceDecl* decl;
+  UseFlags flags;
+  const char* comment;
```

Comment:
I wonder if this needs to become a `std::string` to avoid lifetime issues. We mostly call `ReportXXXUse` with literal strings, but should someone add something like a concatenated string and use `c_str()`, this would refer to a dead value.

In the current flow, that doesn't matter, because the comment will be copied into `OneUse`, and materialized in a new `stdd:string`, but this intermediate storage is a little risky.

Reply:
Good spot. I'll push an update fixing this and the majority of the issues you've raised. I still need to have a go at moving the logic to `IwyuFileInfo` which I'm struggling to find the time for. Maybe later this weekend🤞?

Reply:
Thanks! I've been a little occupied with other things as well, so definitely no rush from my side. Let's percolate on it until it feels stable and nice.